docs/migration.md

```diff
@@ -21,6 +21,53 @@ license: |
 
 # Migration Guide
 
+# Upgrading from 0.5 to 0.6
+- 
+- Since 0.6.0, Celeborn has introduced a new RESTful API namespace: /api/v1, which uses the application/json media type for requests and responses.
+   The `celeborn-openapi-client` SDK is also available to help users interact with the new RESTful APIs.
+   The legacy RESTful APIs have been deprecated and will be removed in future releases.
+   Access the full [RESTful API documentation](./webapi.md) for detailed information.
+  
+  - The mappings of the old RESTful APIs to the new RESTful APIs for Master.
+
+    | Old RESTful API          | New RESTful API                          | Note                                             |
+    |--------------------------|------------------------------------------|--------------------------------------------------|
+    | GET /conf                | GET /api/v1/conf                         |                                                  |
+    | GET /listDynamicConfigs  | GET /api/v1/conf/dynamic                 |                                                  |
+    | GET /threadDump          | GET /api/v1/conf/thread_dump             |                                                  |
+    | GET /applications        | GET /api/v1/applications                 |                                                  |
+    | GET /listTopDiskUsedApps | GET /api/v1/applications/top_disk_usages |                                                  |
+    | GET /hostnames           | GET /api/v1/applications/hostnames       |                                                  |
+    | GET /shuffle             | GET /api/v1/shuffles                     |                                                  |
+    | GET /masterGroupInfo     | GET /api/v1/masters                      |                                                  |
+    | GET /workerInfo          | GET /api/v1/workers                      |                                                  |
+    | GET /lostWorkers         | GET /api/v1/workers                      | get the lostWorkers field in response            |
+    | GET /excludedWorkers     | GET /api/v1/workers                      | get the excludedWorkers field in response        |
+    | GET /shutdownWorkers     | GET /api/v1/workers                      | get the shutdownWorkers filed in response        |
+    | GET /decommissionWorkers | GET /api/v1/workers                      | get the decommissioningWorkers filed in response |
+    | POST /exclude            | POST /api/v1/workers/exclude             |                                                  |
+    | GET /workerEventInfo     | GET /api/v1/workers/events               |                                                  |
+    | POST /sendWorkerEvent    | POST /api/v1/workers/events              |                                                  |
```

Comment:
```suggestion
    | POST /sendWorkerEvent    | POST /api/v1/workers/events              |  the `type` parameter changed to  `eventType`                                               |
```

Reply:
thanks, I think it is better to check the swagger UI for details, here we only show the api mapping.

Reply:
<img width="1695" alt="image" src="https://github.com/user-attachments/assets/f66f0771-b41d-4b02-a178-caabf1fba4d1">

Reply:
The users only check the details in document, not in idea. Right?

Reply:
Use can access the full rest api doc: https://github.com/apache/celeborn/blob/main/docs/webapi.md

For swagger UI:
1. they can check the swagger http://<http-host>:<http-port>/swagger 
2. They can locally preview the OpenAPI specification YAML file in IDE, or use the [Swagger Editor](https://editor.swagger.io/) online for visualization.

Reply:
And here both the request and response are changed, without swagger, it is difficult to describe the API change.

For example: for `/exclude` api:

Before:
```
curl -X POST http://host:port/exclude  -H "Content-Type: application/x-www-form-urlencoded" \
  -d "add=localhost:9200:9201:9202:9203"
```

After:

```
curl -X 'POST' \
  'http://localhost:9098/api/v1/workers/exclude' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "add": [
    {
      "host": "localhost",
      "rpcPort": 9201,
      "pushPort": 9202,
      "fetchPort": 9203,
      "replicatePort": 9204
    }
  ]
}'
```

Reply:
If needed, I can try to generate the docs with openapi-generator, so that we can maintain the RESTapi easily.